### DISCUSSION TITLE
Extract data from fillable form in pdf format

### DISCUSSION THREAD

**Discussion by eriko46** [pymupdf/PyMuPDF#1692] [Looking for help]
Hi , 

I am new to all pymupdf. Trying to scrape a pdf form that has check boxes and fields in it. I am trying to use fitz . What functions would identify the checkbox and the form fields? I would really appreciate if you can get me started with this? 

Thanks from now for your time

**Comment by JorjMcKie** [comment]
Check boxes are a subtype of form fields (widgets).
Look at this example script which walks walks through pages and their widgets and reports all widget infos: https://github.com/pymupdf/PyMuPDF-Utilities/blob/master/fields/list-fields.py

**Reply by eriko46** [reply]
thanks so much for your response. I tried the code but got an "AttributeError: 'Widget' object has no attribute 'widget'" pointing the line "print_widget(a.widget)". Can you give me an idea how to deal with this error? 

thanks again for your time

**Reply by JorjMcKie** [reply]
take this script instead:
[list-fields.zip](https://github.com/pymupdf/PyMuPDF/files/8577737/list-fields.zip)
